sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -308,7 +318,7 @@ static long getEstimatedSizeBytes(Datastore datastore, Query query, @Nullable St
       LOG.info("Latest stats timestamp for kind {} is {}", ourKind, latestTimestamp);
 
       Query.Builder queryBuilder = Query.newBuilder();
-      if (namespace == null) {
+      if (namespace == null || namespace.isEmpty()) {
         queryBuilder.addKindBuilder().setName("__Stat_Kind__");
```

Comment:
Added some comments. Basically `null` or `""` mean default namespace in Datastore.